src/components/Dashboard/RewardsModule.js

```diff
@@ -31,50 +33,72 @@ const useTotalDisputesFees = (arbitrableFees, appealFees) => {
 // Dispute fees + Appeal Fees => DAI fees
 const RewardsModule = React.memo(function RewardsModule({
   rewards,
+  treasury,
   loading,
+  onWithdraw,
   onSettleReward,
   onSettleAppealDeposit,
 }) {
   const wallet = useWallet()
+  const { feeToken } = useCourtConfig()
+
+  const { totalArbitrableFees, totalAppealFees } = useTotalDisputesFees(
+    rewards?.arbitrableFees,
+    rewards?.appealFees
+  )
+
+  // TODO: Handle possible multiple tokens (fee token can change)
+  const treasuryToken = treasury?.find(({ token }) =>
+    addressesEqual(token.id, feeToken.id)
+  )
+  const treasuryBalance = treasuryToken ? treasuryToken.balance : bigNum(0)
+  const totalDisputesFees = totalArbitrableFees.add(totalAppealFees)
+  const totalFees = totalDisputesFees.add(treasuryBalance)
 
   // Form submission
   const handleFormSubmit = async event => {
     event.preventDefault()
 
-    if (!rewards) return
+    const claimRewards = async () => {
+      if (!rewards) return
 
-    try {
-      // Claim all arbitrable fee rewards
-      const arbitrableTxs = []
-      for (const arbitrableFee of rewards.arbitrableFees) {
-        const { disputeId, rounds } = arbitrableFee
-        for (const roundId of rounds) {
-          arbitrableTxs.push(
-            await onSettleReward(disputeId, roundId, wallet.account)
-          )
+      const transactionBag = []
+      try {
+        // Claim all arbitrable fee rewards
+        for (const arbitrableFee of rewards.arbitrableFees) {
+          const { disputeId, rounds } = arbitrableFee
+          for (const roundId of rounds) {
+            transactionBag.push(
+              await onSettleReward(disputeId, roundId, wallet.account)
+            )
+          }
         }
-      }
-
-      await Promise.all(arbitrableTxs.map(tx => tx.wait()))
 
-      // Claim all appeal fee rewards
-      const appealTxs = []
-      for (const appealFee of rewards.appealFees) {
-        const { disputeId, rounds } = appealFee
-        for (const roundId of rounds) {
-          appealTxs.push(await onSettleAppealDeposit(disputeId, roundId))
+        // Claim all appeal fee rewards
+        for (const appealFee of rewards.appealFees) {
+          const { disputeId, rounds } = appealFee
+          for (const roundId of rounds) {
+            transactionBag.push(await onSettleAppealDeposit(disputeId, roundId))
+          }
         }
-      }
 
-      await Promise.all(appealTxs.map(tx => tx.wait()))
-    } catch (err) {
-      console.log(`Error claiming rewards: ${err}`)
+        // Withdraw funds from treasury
+        transactionBag.push(
+          await onWithdraw(feeToken.id, wallet.account, totalFees)
+        )
+
+        await Promise.all(transactionBag.map(tx => tx.wait()))
+      } catch (err) {
+        console.log(`Error claiming rewards: ${err}`)
+      }
     }
+
+    if (!rewards) return
```

Comment:
We already check this in `claimRewards()`, but do we really need to declare the function if all we're going to do is call it immediately? Why not just inline it?

Reply:
We certainly don't need it declared. This was the result of a previous use of `useCallback`